src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+   *                                       attached to it
+   * @param mainGearCircumference          circumference of the main gear
+   * 
+   * @return Returns optional of the util if the gear ratio works with the number
+   *         of max rotations, otherwise returns an empty optional
+   */
+  public static Optional<CRTGearRatioUtil> init(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm,
+      int maxRotations, double mainGearCircumference) {
+    CRTGearRatioUtil u = new CRTGearRatioUtil(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm, maxRotations,
+        mainGearCircumference);
+    if (u.validateGearRatioWithMaxRotations()) { //check if the gear ratio works with the required amount of rotations by the main gear
+      return Optional.of(u);
+    }
+    return Optional.empty();
+  }
+
+  private CRTGearRatioUtil(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm, int maxRotations,
+      double mainGearCircumference) {
+    this.mainGearRotationRatioTerm = (double) mainGearRotationRatioTerm;
+    this.secondaryGearRotationRatioTerm = (double) secondaryGearRotationRatioTerm;
+    this.mainGearRotationRatioTermSimplified = (double) (mainGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.secondaryGearRotationRatioTermSimplfied = (double) (secondaryGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.maxRotations = maxRotations;
+    this.mainGearCircumference = mainGearCircumference;
+  }
+
+  //"These two functions speak for themselves" - Hans Niemann commenting his code
+  public static double gcd(double a, double b) {
+    if (b == 0.0)
+      return a;
+    return gcd(b, a % b);
+  }
+
+  public static double mod(double a, double b) {
+    double r = a % b;
+    if (r < 0) {
+      r += b;
+    }
+    return r;
+  }
+
+  /**
+   * 
+   * @return Returns a multiplier used by CRT. You shouldn't need to use this function yourself, and
+   *         I don't fully understand it (Victor Chen did the math behind it).
+   */
+  public int kFactor() {
+    double sum = 0;
+    double a = secondaryGearRotationRatioTermSimplfied;
+    double b = mainGearRotationRatioTermSimplified;
+    for (int i = 1; i < b; i++) {
+      sum += ((a * (b - i) - 1.0) / b)
+          * (1 + Math.floor(((a * (b - i) - 1.0) / b)) - Math.ceil(((a * (b - i) - 1.0) / b)));
+    }
+    return (int) ((int) ((sum * b) + 1.0) / a); //basically, when you apply CRT, you can multiply by a certain factor (what this function returns, a constant for a given gear ratio) such that when using modding that factor times the output of crt and the simplified main gear rotation ratio term, it will correctly output a number between 0 and maxrotations. then, to account for wrap around and to make 0 full rotations work correctly, mod the main gear ratio term by the output + the simplified term
```

Comment:
Is the behavior of `(int)` correct for you? Or do you want to round? https://stackoverflow.com/a/2143505

Reply:
In this case it shouldn't matter as the way Victor wrote it makes it always be an integer anyway. The int cast it just to make the return type correct.